app/components/access_panels/other_listings_component.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module AccessPanels
+  class OtherListingsComponent < AccessPanels::Base
+    def render?
+      earthworks_url.present? || purls.present?
+    end
+
+    def earthworks_url
+      @earthworks_url ||= construct_earthworks_url
+    end
+
+    # @return [Array<String>] URLs representing links to PURL, whether specified in MARC or MODS, and de-duped, e.g. ["https://purl.stanford.edu/hj948rn6493"]
+    def purls
+      @purls ||= (mods_other_locations + marc_other_locations).uniq.find_all { |str| str.start_with?(Settings.PURL_EMBED_RESOURCE) }
+    end
+
+    # @return [Array<String>] a tags representing PURL links
+    def purl_links
+      # text of the link is the actual URL if there is more than one unique URL, so there aren't multiple links with the same title
+      return purls.map { |purl| link_to(purl, purl) } if purls.size > 1
+
+      # returned as an array for type consistency
+      Array(link_to('Stanford Digital Repository', purls.first))
+    end
+
+    private
+
+    # @return [Array<String>] URLs representing links to other locations, extracted from MODS metadata, e.g. ["https://purl.stanford.edu/hj948rn6493"]
+    def mods_other_locations
+      #   latter of those is used in app/views/catalog/record/_mods_bibliographic.html.erb (but has some clutter unlike the mockup, for this purpose)
+      Array(document['url_fulltext'])
+    end
+
+    # @return [Array<String>] URLs representing links to other locations, extracted from MARC metadata, e.g. ["https://purl.stanford.edu/hj948rn6493"]
+    def marc_other_locations
+      # the fetch with the default empty hash prevents errors for data without MARC representation
+      Array(document.fetch('marc_links_struct', {}).pluck('href'))
+    end
+
+    def construct_earthworks_url
+      return nil unless document.schema_dot_org?
+
+      # This value should always be a single hash.  It's in the required JSON-LD format for a value of type DataCatalog as
+      # specified by schema.org, even if there is a single value. It should be safe to assume the value won't be a list (despite
+      # the seemingly generalizable '@type' and 'name' fields in the hash).
+      data_catalog_name = document.as_schema_dot_org.dig('includedInDataCatalog', 'name')
```

Comment:
Pulling this out of `as_schema_dot_org` seems a little indirect (and surpising) to me. Can we just use the type `geo` like the indexer does?

https://github.com/sul-dlss/searchworks_traject_indexer/blob/main/lib/traject/config/sdr_config.rb#L351

Can we check that here, or

Reply:
thanks!

i only went this direction because it was the suggestion on the ticket, and because i'm pretty unfamiliar with searchworks and its indexing.  but your suggestion worked in my local testing, definitely feels less circuitous, and i think alleviates a spec difficulty i was running into.  (i was having trouble indexing a harvested fixture with a `schema_dot_org_struct` using `bin/rails searchworks:fixtures`, and i had to submit JSON directly through the solr web UI to get a document for local testing).

Reply:
Sounds good. I think either using type `geo` or explicitly indexing release tags (if there's some reason type geo is problematic) is probably the way to go.

Reply:
would the latter be a searchworks_traject_indexer change?  any reason to investigate that and block merging this before figuring more out?  i don't really have any sense one way or the other of whether or how using type geo could cause issues.

Reply:
Hmm. I think this actually may be problematic. There are many thousands of items with content type `map` (not `geo`) that are released to EarthWorks, and we allow quite a few other types too. There's also no guarantee that any given `geo` item is actually in EarthWorks. We might need to change the implementation (both here and for schema.org metadata) to look at release tags.

Reply:
@cbeer is it a bad idea to ask purl-fetcher at render time about release info?